rustls/src/client/client_conn.rs

```diff
@@ -215,6 +216,18 @@ pub struct ClientConfig {
 
     /// How to verify the server certificate chain.
     pub(super) verifier: Arc<dyn verify::ServerCertVerifier>,
+
+    /// How to decompress the server's certificate chain.
+    ///
+    /// If this is non-empty, the [RFC8779] certificate compression
+    /// extension is offered, and any compressed certificates are
+    /// transparently decompressed during the handshake.
+    ///
+    /// This only applies to TLS1.3 connections.  It is ignored for
+    /// TLS1.2 connections.
+    ///
+    /// [RFC8779]: https://datatracker.ietf.org/doc/rfc8879/
+    pub cert_decompressors: Vec<&'static dyn compress::CertDecompressor>,
```

Comment:
Maybe this could/should be `Arc<[..]>` instead?

Reply:
I feel this makes the common uses (like adding a value, or removing one) pretty wordy? (eg, converting to a vec, adding a value, then using Vec's `Into<Arc<[T]>>`) -- maybe I missed something though.